lib/dry/container/mixin.rb

```diff
@@ -179,6 +179,22 @@ def each_key(&block)
         self
       end
 
+      # Calls block once for each key/value pair in the container, passing the key and the registered item parameters.
+      #
+      # If no block is given, an enumerator is returned instead.
+      #
+      # @return [Dry::Container::Mixin] self
+      #
+      # @api public
+      #
+      # @note In discussions with other developers, it was felt that being able to iterate over not just
+      #       the registered keys, but to see what was registered would be very helpful. This is a step
+      #       toward doing that.
+      def each_pair(&block)
+        config.resolver.each_pair(_container, &block)
+        self
```

Comment:
Still not sure on the feature as a whole yet, but I think that returning `self` here is a bad idea, it doesn't seem like a good candidate for method chaining and encourages mutation in some form or another